src/NuGet.Core/NuGet.Protocol/Plugins/PluginManager.cs

```diff
@@ -119,7 +119,7 @@ public async Task<IEnumerable<PluginCreationResult>> CreatePluginsAsync(
             var pluginCreationResults = new List<PluginCreationResult>();
 
             // Fast path
-            if (source.PackageSource.IsHttp && IsPluginPossiblyAvailable())
+            if (source.PackageSource.IsHttp)
```

Comment:
Doesn't this lead to us fetching the ServinceIndexResourceV3 even where there are no plugins available? 
Digging more, I'm wondering if we'd ever try to initialize a download plugin unless it's specified with an env var (which is probably a good thing tbh).

Reply:
This logic may need reworked to ensure we're not fetching resources or starting plugins too aggressively.